File: Frontend/api-tester/src/app/models/api-request.model.ts
export interface ApiRequestModel {
    url: string;
    method: string;
    headers: { [key: string]: string };
    requestBody: string;
    contentType: string;
    timeoutSeconds: number;
    authType: string;
    username?: string;
    password?: string;
    bearerToken?: string;
    apiKeyName?: string;
    apiKeyValue?: string;
    cronExpression?: string;
  }

File: Backend/ApiTester.Api/ApiTester.Api/Models/ApiRequestModel.cs
using System.ComponentModel.DataAnnotations;

namespace ApiTester.Api.Models
{
    public class ApiRequestModel
    {
        [Required(ErrorMessage = "URL is required")]
        public string? Url { get; set; }

        [Required]
        public string Method { get; set; } = "GET";

        public Dictionary<string, string> Headers { get; set; } = new Dictionary<string, string>();

        public string? RequestBody { get; set; }

        public string ContentType { get; set; } = "application/json";

        [Range(1, 300, ErrorMessage = "Timeout must be between 1 and 300 seconds")]
        public int TimeoutSeconds { get; set; } = 30;

        public string? AuthType { get; set; }

        public string? Username { get; set; }

        public string? Password { get; set; }

        public string? BearerToken { get; set; }

        public string? ApiKeyName { get; set; }

        public string? ApiKeyValue { get; set; }
    }
}